### DISCUSSION TITLE
New helper for conditionally redirecting

### DISCUSSION THREAD

**Discussion by jase-languasco** [laravel/framework#52423] [Ideas]
I'd like to suggest the addition of `redirect_if` and `redirect_unless`

It's possible it doesn't already exist because of the number of arguments that are required by `redirect()`, otherwise this feels like a good candidate for a new `_if` helper

**Comment by shaedrich** [comment]
If that were to be implemented into the `Redirect` facade, this could be implemented like `Redirect::if()->to()` and work like the `optional()` helper 🤔

**Reply by henzeb** [reply]
to follow conventions, it would be when / unless. which is is simply adding a Conditionable trait....

**Reply by shaedrich** [reply]
Great, couldn't be easier, then 👍🏻
